File: BE/health-service/src/main/java/com/s206/health/exercise/dto/response/WeeklyExerciseStatsResponse.java
package com.s206.health.exercise.dto.response;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

import java.time.LocalDate;
import java.util.List;

@Getter
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class WeeklyExerciseStatsResponse {
    private List<WeeklyExercise> weeklyExercises;

    @Getter
    @Builder
    public static class WeeklyExercise {
        private LocalDate startDate;
        private LocalDate endDate;
        private Double avgDailyCalories;
        private Integer totalExerciseCount;
        private Integer totalCalories;
    }
}

File: FE/mobile/src/main/java/com/example/diaviseo/network/exercise/dto/res/WeeklyExerciseStatsResponse.kt
package com.example.diaviseo.network.exercise.dto.res

data class WeeklyExerciseStatsResponse(
    val weeklyExercises: List<WeeklyExercise>
) {
    data class WeeklyExercise(
        val startDate: String,
        val endDate: String,
        val avgDailyCalories: Double,
        val totalExerciseCount: Int?,
        val totalCalories: Int?
    )
}